packages/components/src/custom-select-control/index.js

```diff
@@ -141,7 +141,14 @@ export default function CustomSelectControl( {
 					className="components-custom-select-control__button-icon"
 				/>
 			</Button>
-			<ul { ...menuProps }>
+			{ /* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */ }
+			<ul
+				{ ...menuProps }
+				onKeyDown={ ( e ) => {
+					e.stopPropagation();
+					menuProps.onKeyDown( e );
```

Comment:
I'm not sure we're guaranteed that `onKeyDown` is always defined on `menuProps` — it's probably safer to use optional chaining?

```suggestion
					menuProps?.onKeyDown?.( e );
```